pennylane/ops/op_math/decompositions/single_qubit_unitary.py

```diff
@@ -379,7 +379,18 @@ def _xzx_decomposition(U, wire, return_global_phase=False):
      GlobalPhase(1.1759220332464762, wires=[])]
 
     """
+    phis, thetas, lams, gammas = _get_xzx_angles(U)
 
+    operations = [qml.RX(lams, wire), qml.RZ(thetas, wire), qml.RX(phis, wire)]
+    if return_global_phase:
+        operations.append(qml.GlobalPhase(-gammas))
+
+    return operations
```

Comment:
Great idea to break this up.

Reply:
I've done the same in my PR here: https://github.com/PennyLaneAI/pennylane/pull/7211, maybe this PR can reuse some of the code changes in my PR.

Reply:
Given the plan to move quick with this, I'd rather avoid too much dependence on product-facing PRs. Do you think your PR is going in soon? I suspect we will merge this in the next day or so if not.

Reply:
Chatted with Lillian offline and decided to not rebase onto my branch